### ISSUE TITLE
Checkboxes in tree view need aria-labelledby (see notes)

### ISSUE THREAD

**Issue by mathisscott** [vmware/clarity#1926]
URL: http://clarity-ks.surge.sh/treeviews

Tools: Web Accessibility Toolbar (WAT2.0) / Inspect32

Reproduced Steps:
1. Login Clarity Kitchen Sink with web browser.
2. Go to Tree Views tab.
3. Check the name property of all check boxes in Tree Views tab.

Actual Results:
The name property of all check boxes is “”.  
It’s difficult to distinguish between them for Blind.
For more details, please refer to “TreeViews-NamePropertyIsEmpty.png”.

Expected Results:
The name property should have a descriptive and unique name property.

Note:
This issue also occurs on Vertical Nav tab.

Violation of Section 508 Standards: 
1194.21(d) Sufficient information about a user interface element including the identity, operation and state of the element shall be available to assistive technology. When an image represents a program element, the information conveyed by the image must also be available in text. 
User Impact: Blind.

Screen shot:
1.	TreeViews-NamePropertyIsEmpty.png

![image.png](https://images.zenhubusercontent.com/582f4cffb9dc3c6220222f57/f5aeec79-2e7e-4512-97b6-bc09a641dec5)

NOTE: this may also be an issue on the website

**Comment by youdz** [comment]
The `name` attribute on radio buttons and checkboxes isn't used for accessibility purposes. It's not read by screenreaders, as the whole point of it is to generate the correct query parameter in the URL (or POST data) when submitting the form.

I believe what we're missing here is an `aria-labelledby` attribute on the checkboxes, pointing to the text in the tree node.

**Comment by mathisscott** [comment]
For NG components, need to make sure we tweak the checkboxes (for tree nodes) so that they have unique identifiers.
